### PR TITLE
22542 add rails 5.2 support

### PR DESCRIPTION
**Pull request by Altivi**
**Link to Acunote task**:

- [Task 22542](https://integracredit.acunote.com/tasks/22542)

**Description of changes this PR makes**:
- Updated to support rails 5.2.

### CODE REVIEW

File: lib/moribus/extensions/has_current_extension.rb [hunk 1]
```diff
@@ -11,14 +11,12 @@ def remove_target!(*)
...
           # Use custom update to avoid running ActiveRecord optimistic locking
           # and to avoid updating lock_version column:
-          klass          = target.class
-          is_current_col = klass.columns.detect { |c| c.name == "is_current" }
-          id_column      = klass.columns.detect { |c| c.name == klass.primary_key }
+          klass = target.class
 
           sql =  "UPDATE #{klass.quoted_table_name} " \
-                 "SET \"is_current\" = #{klass.quote_value(false, is_current_col)} "
+                 "SET \"is_current\" = #{klass.connection.quote(false)} "
```
Comment (zbelzer) [inline lib/moribus/extensions/has_current_extension.rb]: Can you provide a little more background on this change?

Comment (Altivi) [reply lib/moribus/extensions/has_current_extension.rb]:
```
NoMethodError: 
    undefined method `quote_value' for #<Class:0x007fc4aaf01ba0>
```
The last version when the `quote_value` method appeared in `activerecord` repo was 3.1 https://github.com/rails/rails/blob/3-1-stable/activerecord/lib/active_record/base.rb#L850 So I replaced it with rails 5.2 method `quote` https://github.com/rails/rails/blob/aeba121a83965d242ed6d7fd46e9c166079a3230/activerecord/lib/active_record/connection_adapters/abstract/quoting.rb#L11

Comment (albertosaurus) [reply lib/moribus/extensions/has_current_extension.rb]: Then how was it working in 4.2?

Comment (albertosaurus) [reply lib/moribus/extensions/has_current_extension.rb]: Then how was it working in 4.2?

**Comment by zbelzer** [comment]
I was wondering too but got distracted. Will double check

**Comment by zbelzer** [comment]
Application tests pass in 4.2 with this code. I don't think there's a backwards compatibility problem.

File: lib/moribus/extensions/delegate_associated.rb [hunk 2]
```diff
@@ -33,7 +33,7 @@ module DelegateAssociated
       # presented in Customer, the code will result in exception without
       # the following hook:
       def column_for_attribute(name)
-        unless (column = super).nil?
+        unless (column = super).nil? || column.is_a?(ActiveRecord::ConnectionAdapters::NullColumn)
```
Comment (albertosaurus) [inline lib/moribus/extensions/delegate_associated.rb]: `ActiveRecord::ConnectionAdapters::NullColumn` is undefined in Rails 4.2.

**Comment by albertosaurus** [comment]
I'm pretty certain this isn't fully compatible with 4.2. @zbelzer I'm going to up the minimum version required to 5.0 unless you're strongly opposed.

File: lib/moribus/extensions/delegate_associated.rb (hunk 2 above)
Comment (zbelzer) [reply lib/moribus/extensions/delegate_associated.rb]: I still forget about lazy/conditional constant evaluation.

**Pull request merged**